components/draw/PointOrPolygonMode.svelte

```diff
@@ -0,0 +1,55 @@
+<script>
+  import {
+    map,
+    gjScheme,
+    newFeatureId,
+    setCurrentlyEditing,
+  } from "../../stores.js";
+
+  export let mode;
+  // "point" or "polygon"
+  export let type;
+  export let drawControls;
+  // Maybe null at first
+  export let routeSnapper;
+
+  $: {
+    if (mode == type) {
+      drawControls.changeMode(`draw_${type}`);
+      routeSnapper?.stop();
```

Comment:
Here's another kind of weird state-changing example. Anytime we enter a mode, we have to make sure to turn off stuff from every other mode. This feels very "leaky", not encapsulated at all.